DAO/project/contracts/DAO-contract/tests/functions/core/create_proposal.rs

```diff
@@ -14,20 +17,30 @@ mod success {
         constructor(&deployer.dao_voting, gov_token_id).await;
 
         let proposal_transaction = proposal_transaction(gov_token_id);
-        create_proposal(&user.dao_voting, 10, 10, proposal_transaction.clone()).await;
+        let response =
+            create_proposal(&user.dao_voting, 10, 10, proposal_transaction.clone()).await;
+
+        let log = response.get_logs_with_type::<CreatePropEvent>().unwrap();
+        let event = log.get(0).unwrap();
+
+        let expected_proposal = ProposalInfo {
+            author: Identity::Address(user.wallet.address().into()),
+            yes_votes: 0,
+            no_votes: 0,
+            acceptance_percentage: 10,
+            proposal_transaction: proposal_transaction.clone(),
+            deadline: 14,
+            executed: false,
+        };
 
         assert_eq!(
-            proposal(&user.dao_voting, 0).await,
-            ProposalInfo {
-                author: Identity::Address(user.wallet.address().into()),
-                yes_votes: 0,
-                no_votes: 0,
-                acceptance_percentage: 10,
-                proposal_transaction,
-                deadline: 14,
-                executed: false,
+            *event,
+            CreatePropEvent {
```

Comment:
```suggestion
            CreateProposalEvent {
```
https://github.com/FuelLabs/sway-applications/blob/master/DAO/project/contracts/DAO-contract/src/events.sw#L7